source/app-development/tutorial-ps-to-quota.rst

```diff
@@ -0,0 +1,451 @@
+        end
+      end
+
+After the changes part of the command.rb will look like this:
+
+.. code:: ruby
+
+    class Command
+      def to_s
+        "quota -spw"
+      end
+
+      Quota = Struct.new(:filesystem, :blocks, :blocks_quota, :blocks_limit, :blocks_grace, :files, :files_quota, :files_limit, :files_grace)
+
+      # Parse a string output from the `ps aux` command and return an array of
+      # AppProcess objects, one per process
+      def parse(output)
+        lines = output.strip.split("\n")
+        lines.drop(2).map do |line|
+          Quota.new(*(line.split))
+        end
+      end
+
+Now when we run the test they pass:
+
+.. code:: sh
+
+    $ rake
+    Run options: --seed 60317
+
+    # Running:
+
+    .
+
+    Finished in 0.000966s, 1035.1494 runs/s, 6210.8963 assertions/s.
+
+    1 runs, 6 assertions, 0 failures, 0 errors, 0 skips
+
+Update app.rb and view/index.html
+.................................
+
+Update app.rb:
+
+.. code:: diff
+
+    helpers do
+      def title
+    -    "Passenger App Processes"
+    +    "Quota"
+      end
+    end
+
+    # Define a route at the root '/' of the app.
+    get '/' do
+      @command = Command.new
+    -  @processes, @error = @command.exec
+    +  @quotas, @error = @command.exec
+
+      # Render the view
+      erb :index
+    end
+
+
+In view/index.html, replace the table with this:
+
+.. code:: erb
+
+    <table class="table table-bordered">
+      <tr>
+        <th>Filesystem</th>
+        <th>Blocks</th>
+        <th>Blocks Quota</th>
+        <th>Blocks Limit</th>
+        <th>Blocks Grace</th>
+        <th>Files</th>
+        <th>Files Quota</th>
+        <th>Files Limit</th>
+        <th>Files Grace</th>
+      </tr>
+      <% @quotas.each do |quota| %>
+      <tr>
+        <td><%= quota.filesystem %></td>
+        <td><%= quota.blocks %></td>
+        <td><%= quota.blocks_quota %></td>
+        <td><%= quota.blocks_limit %></td>
+        <td><%= quota.blocks_grace %></td>
+        <td><%= quota.files %></td>
+        <td><%= quota.files_quota %></td>
+        <td><%= quota.files_limit %></td>
+        <td><%= quota.files_grace %></td>
+      </tr>
+      <% end %>
+    </table>
+
+These changes should not require an app restart. Go to the launched app and reload the page to see the changes.
+
+Brand App
+---------
+
+The app is looking good, but the details page still shows the app title "Passenger App Processes". To change this and the icon, edit the manifest.yml:
+
+.. code:: diff
+
+    -name: Passenger App Processes
+    -description: Display your running Passenger app proceseses in a table
+    +name: Quota
+    +description: Display quotas
+    +icon: fa://hdd-o
+
+* The icon follows format of ``fa://{FONTAWESOMENAME}`` where you replace ``{FONTAWESOMENAME}`` with an icon from http://fontawesome.io/icons/.
+  In this case we are using ``fa-hdd-o`` which we write in the manifest as ``fa://hdd-o``: http://fontawesome.io/icon/hdd-o/
```

Comment:
This looks a little funny in the build. Were you trying to hyperlink `fa-hdd-o` to the fontawesome site? The code looks like you were going for a rst hyperlink and the link seems kind of appendageal at the end.

Reply:
![fa](https://user-images.githubusercontent.com/2374718/30763756-c0f432a2-9fb5-11e7-9df9-eacb411a6350.png)